ipa-core/src/protocol/ipa_prf/mod.rs

```diff
@@ -721,3 +721,109 @@ pub mod tests {
         });
     }
 }
+
+#[cfg(all(test, all(feature = "compact-gate", feature = "in-memory-infra")))]
+mod compact_gate_tests {
+
+    use ipa_step::StepNarrow;
+
+    use crate::{
+        ff::{
+            boolean_array::{BA20, BA5, BA8},
+            U128Conversions,
+        },
+        helpers::query::DpMechanism,
+        protocol::{
+            ipa_prf::{oprf_ipa, oprf_padding::PaddingParameters},
+            step::{ProtocolGate, ProtocolStep},
+        },
+        test_executor::run,
+        test_fixture::{ipa::TestRawDataRecord, Reconstruct, Runner, TestWorld, TestWorldConfig},
+    };
+
+    #[test]
+    fn saturated_agg() {
+        const EXPECTED: &[u128] = &[0, 255, 255, 0, 0, 0, 0, 0];
```

Comment:
if we're trying to test going from 2 bytes to 4, shouldn't we test some much bigger numbers?